### ISSUE TITLE
Uncaught ReferenceError: Buffer is not defined

### ISSUE THREAD

**Issue by navan101** [davidmyersdev/vite-plugin-node-polyfills#22]
Hi @davidmyersdev,

1. When i use vite-plugin-node-polyfills and config default, run dev environment it gives error buffer.

<img width="1422" alt="Screen Shot 2023-06-20 at 16 22 15" src="https://github.com/davidmyersdev/vite-plugin-node-polyfills/assets/7380005/44d7582d-b286-47cc-ba04-02760698f6de">

![Screen Shot 2023-06-20 at 16 22 32](https://github.com/davidmyersdev/vite-plugin-node-polyfills/assets/7380005/4176dc60-4c71-417d-8432-f65c07c223c6)

2. When i run the build environment it works.

![Screen Shot 2023-06-20 at 16 23 22](https://github.com/davidmyersdev/vite-plugin-node-polyfills/assets/7380005/320f9ca6-fd4d-42a0-8ecf-39723722d5f4)

link: [https://codesandbox.io/p/github/navan101/vite-buffer/master?file=/src/App.vue:1,1&workspaceId=89d4ce07-40e9-48ee-965a-5e9183e6320b]

Thank you!

**Comment by davidmyersdev** [comment]
@navan101 this is fixed in `v0.11.0`.

**Comment by smart-rfe** [comment]
Hi @davidmyersdev,

I seem to have the same kind of problem, even with version 0.11.0.

![Capture d’écran 2023-08-08 à 16 55 58](https://github.com/davidmyersdev/vite-plugin-node-polyfills/assets/86787797/f1a00f65-2986-462a-a297-98db10e35b01)

![Capture d’écran 2023-08-08 à 16 56 40](https://github.com/davidmyersdev/vite-plugin-node-polyfills/assets/86787797/0088ee50-d84f-4665-8cde-034c8a73ca69)


```require('buffer')``` gives undefined so ```kMaxLength.toString(16)``` is impossible.


I'm in the same situation as @navan101, I have the default configuration and the build passes but the dev environment gives me this error.

Is this something you can fix on your end?

**Comment by VysockyStan** [comment]
> Hi @davidmyersdev,
> 
> I seem to have the same kind of problem, even with version 0.11.0.
> 
> ![Capture d’écran 2023-08-08 à 16 55 58](https://user-images.githubusercontent.com/86787797/259142852-f1a00f65-2986-462a-a297-98db10e35b01.png)
> 
> ![Capture d’écran 2023-08-08 à 16 56 40](https://user-images.githubusercontent.com/86787797/259142844-0088ee50-d84f-4665-8cde-034c8a73ca69.png)
> 
> `require('buffer')` gives undefined so `kMaxLength.toString(16)` is impossible.
> 
> I'm in the same situation as @navan101, I have the default configuration and the build passes but the dev environment gives me this error.
> 
> Is this something you can fix on your end?

I have the same issue

**Comment by davidmyersdev** [comment]
@rfe-sdev @VysockyStan @navan101 can any of you confirm whether this is working properly in `v0.11.1`? Also, are you all using `require` rather than `import`? This could be an issue with the CommonJS build if so.

**Comment by JounQin** [comment]
<img width="677" alt="image" src="https://github.com/davidmyersdev/vite-plugin-node-polyfills/assets/8336744/bf2d7708-2a78-4c64-a949-78edd47839dd">

<img width="880" alt="image" src="https://github.com/davidmyersdev/vite-plugin-node-polyfills/assets/8336744/1d61a13c-163e-4cb5-8a4a-d9cf1355149e">


Same issue here

**Comment by VysockyStan** [comment]
@davidmyersdev I've tried 0.11.1. The issue persists. It occurs in 3rd-party lib which which uses browserify-zlib as on the screen that @rfe-sdev posted above. As you may see it uses require. So there is no chance to make it work with CommonJS ?

**Comment by VysockyStan** [comment]
But the difference that in my case Buffer is not undefined, however kMaxLength is

**Comment by aryzing** [comment]
I may have opened a dupe of this issue, https://github.com/davidmyersdev/vite-plugin-node-polyfills/issues/34. I also created a [reproduction repo](https://github.com/aryzing/repro-node-polyfill-not-working) to help with debugging.

**Comment by VysockyStan** [comment]
I'm using [https://www.npmjs.com/package/@ringcentral/sdk](url) in React, which produces the error

**Comment by aryzing** [comment]
I've noticed that when removing the import that needs `Buffer`, the app does run fine, and `Buffer` is defined in the browser's console. This leads me to believe that perhaps there's something about the plugin that causes the polyfills to be loaded too late?

**Comment by davidmyersdev** [comment]
@aryzing that does appear to be the case. I think the code that is erroring is actually a _side effect_ that runs during the import process (before the polyfill code gets a chance to run). I'm trying to figure out a way to get this working though.

**Comment by davidmyersdev** [comment]
I was able to verify that both reproductions (thank you to @navan101 and @aryzing for providing them) have been fixed in `v0.11.2`.

**Comment by aryzing** [comment]
So some good news, and some bad news. Good news: for smaller projects, `v0.11.2` does work. The bad news is that for a larger project I'm working on, this plugin is causing the build to fail with a heap limit error,

```
FATAL ERROR: Reached heap limit Allocation failed - JavaScript heap out of memory
```

<details>
<summary>Stack trace</summary>

```
 1: 0xc98550 node::Abort() [/home/user/.nvm/versions/node/v20.5.1/bin/node]
 2: 0xb700a7  [/home/user/.nvm/versions/node/v20.5.1/bin/node]
 3: 0xebae80 v8::Utils::ReportOOMFailure(v8::internal::Isolate*, char const*, v8::OOMDetails const&) [/home/user/.nvm/versions/node/v20.5.1/bin/node]
 4: 0xebb167 v8::internal::V8::FatalProcessOutOfMemory(v8::internal::Isolate*, char const*, v8::OOMDetails const&) [/home/user/.nvm/versions/node/v20.5.1/bin/node]
 5: 0x10cc7d5  [/home/user/.nvm/versions/node/v20.5.1/bin/node]
 6: 0x10e4658 v8::internal::Heap::CollectGarbage(v8::internal::AllocationSpace, v8::internal::GarbageCollectionReason, v8::GCCallbackFlags) [/home/user/.nvm/versions/node/v20.5.1/bin/node]
 7: 0x10ba771 v8::internal::HeapAllocator::AllocateRawWithLightRetrySlowPath(int, v8::internal::AllocationType, v8::internal::AllocationOrigin, v8::internal::AllocationAlignment) [/home/user/.nvm/versions/node/v20.5.1/bin/node]
 8: 0x10bb905 v8::internal::HeapAllocator::AllocateRawWithRetryOrFailSlowPath(int, v8::internal::AllocationType, v8::internal::AllocationOrigin, v8::internal::AllocationAlignment) [/home/user/.nvm/versions/node/v20.5.1/bin/node]
 9: 0x1098e76 v8::internal::Factory::NewFillerObject(int, v8::internal::AllocationAlignment, v8::internal::AllocationType, v8::internal::AllocationOrigin) [/home/user/.nvm/versions/node/v20.5.1/bin/node]
10: 0x14f3ca6 v8::internal::Runtime_AllocateInYoungGeneration(int, unsigned long*, v8::internal::Isolate*) [/home/user/.nvm/versions/node/v20.5.1/bin/node]
11: 0x7f93f7ed9ef6
Aborted (core dumped)
```

</details>

Possibly useful info:

- As modules get processed, the speed at which they get processed decreases.
- With Node v18 Vite processes ~750 modules before crashing, with Node v20 closer to ~1100
- I'm developing a web extension, which requires assets to be written to disk even in dev mode. This is done via `vite-plugin-web-extension`.

I've created a [new reproduction repo](https://github.com/aryzing/repro-node-polyfills-large-repo) using this plugin on a large codebase which displays the error above during the build.
